### PR TITLE
Change type aliases into functions that substitute parameters

### PR DESCRIPTION
**Pull request by Baransu**
Closes: #31

### CODE REVIEW

File: src/ExAlias.elm [hunk 1]
```diff
@@ -12,29 +13,55 @@ getAliases list =
...
+handleAlias args t =
+    case t of
+        (TypeConstructor name vars) :: rest ->
+            let
+                v =
+                    zip args vars
+                        |> List.map
+                            (\( a, v ) ->
+                                case a of
+                                    (TypeConstructor [ name ] []) as t ->
```
Comment (wende) [inline src/ExAlias.elm]: That definitely needs refactoring

File: src/Helpers.elm [hunk 2]
```diff
@@ -172,6 +173,18 @@ translateOperator op =
...
+
+
+zip : List a -> List b -> List ( a, b )
+zip xs ys =
+    case ( xs, ys ) of
+        ( x :: xBack, y :: yBack ) ->
+            ( x, y ) :: zip xBack yBack
+
+        ( _, _ ) ->
+            []
```
Comment (wende) [inline src/Helpers.elm]: Does that differ from List.zip?

Comment (Baransu) [inline src/Helpers.elm]: There is no `List.zip` :/

Comment (wende) [inline src/Helpers.elm]: `List.map2 (,)`

**Comment by wende** [comment]
screen shot 2017-05-07 at 19 17 26

https://www.youtube.com/watch?v=FaWhXXdQkvk

**Comment by wende** [comment]
Too much changed. There is no sense in continuing this PR. It's better rewritten from scratch

**Pull request closed without merging**